azure/blobs_logs_monitoring/index.js

```diff
@@ -2,191 +2,602 @@
     }
+}
+
+class ScrubberRule {
+    constructor(name, pattern, replacement) {
+        this.name = name;
+        this.replacement = replacement;
+        this.regexp = RegExp(pattern, 'g');
+    }
+}
+
 
-    var socket = getSocket(context);
-    var sender = tagger => record => {
-        record = tagger(record, context);
-        if (!send(socket, record)) {
-            // Retry once
-            socket = getSocket(context);
-            send(socket, record);
+class Batcher {
+    constructor(context, maxItemSizeBytes, maxBatchSizeBytes, maxItemsCount) {
+        this.maxItemSizeBytes = maxItemSizeBytes;
+        this.maxBatchSizeBytes = maxBatchSizeBytes;
+        this.maxItemsCount = maxItemsCount;
+    }
+
+    batch(items) {
+        var batches = [];
+        var batch = [];
+        var sizeBytes = 0;
+        var sizeCount = 0;
+        for (var i = 0; i < items.length; i++) {
+            var item = items[i];
+            var itemSizeBytes = this.getSizeInBytes(item);
+            if (
+                sizeCount > 0 &&
+                (sizeCount >= this.maxItemsCount ||
+                    sizeBytes + itemSizeBytes > this.maxBatchSizeBytes)
+            ) {
+                batches.push(batch);
+                batch = [];
+                sizeBytes = 0;
+                sizeCount = 0;
+            }
+            // all items exceeding maxItemSizeBytes are dropped here
+            if (itemSizeBytes <= this.maxItemSizeBytes) {
+                batch.push(item);
+                sizeBytes += itemSizeBytes;
+                sizeCount += 1;
+            }
         }
-    };
 
-    var logs;
-    if (typeof blobContent === 'string') {
-        logs = blobContent.trim().split('\n');
-    } else if (Buffer.isBuffer(blobContent)) {
-        logs = blobContent
-            .toString('utf8')
-            .trim()
-            .split('\n');
-    } else {
-        logs = JSON.stringify(blobContent)
-            .trim()
-            .split('\n');
+        if (sizeCount > 0) {
+            batches.push(batch);
+        }
+        return batches;
     }
 
-    logs.forEach(log => {
-        handleLogs(sender, log, context);
-    });
+    getSizeInBytes(string) {
+        if (typeof string !== 'string') {
+            string = JSON.stringify(string);
+        }
+        return Buffer.byteLength(string, 'utf8');
+    }
+}
 
-    socket.end();
-    context.done();
-};
+class HTTPClient {
+    constructor(context) {
+        this.context = context;
+        this.httpOptions = {
+            hostname: DD_HTTP_URL,
+            port: DD_HTTP_PORT,
+            path: '/v1/input',
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'DD-API-KEY': DD_API_KEY
+            }
+        };
+        this.scrubber = new Scrubber(this.context, SCRUBBER_RULE_CONFIGS);
+        this.batcher = new Batcher(
+            this.context,
+            256 * 1000,
```

Comment:
How these limits were decided? Is it based on the doc [here](https://docs.datadoghq.com/api/latest/logs/#send-logs)

Reply:
Uncertain they were added in a 2021 PR for the event hub forwarder
https://github.com/DataDog/datadog-serverless-functions/pull/456